13-dplyr.Rmd

```diff
@@ -0,0 +1,578 @@
+---
+layout: page
+title: R for reproducible scientific analysis
+subtitle: Dataframe manipulation with dplyr
+minutes: 90
+---
+
+```{r, include=FALSE}
+source("tools/chunk-options.R")
+opts_chunk$set(fig.path = "fig/12-plyr-")
+# Silently load in the data so the rest of the lesson works
+gapminder <- read.csv("data/gapminder-FiveYearData.csv", header=TRUE)
+```
+
+> ## Learning objectives {.objectives}
+>
+> * To be able to use the 6 main dataframe manipulation 'verbs' with pipes in `dplyr`
+>
+
+Manipulation of dataframes means many things to many researchers, we often select certain observations (rows) or variables (columns), we often group the data by a certain variable(s), or we even calculate summary statistic. We can do these operations using the normal base R operations:
+
+```{r}
+mean(gapminder[gapminder$continent == "Africa", "gdpPercap"])
+mean(gapminder[gapminder$continent == "Americas", "gdpPercap"])
+mean(gapminder[gapminder$continent == "Asia", "gdpPercap"])
+```
+
+But this isn't very *nice* because there is a fair bit of repetition. Repeating yourself will cost you time, both now and later, and potentially introduce some nasty bugs.
+
+#### The `dplyr` package
+
+Luckily, the [`dplyr`](https://cran.r-project.org/web/packages/dplyr/dplyr.pdf) package provides a number of very useful functions for manipulating dataframes in a way that will reduce the above repetition, reduce the probability of making errors, and probably even save you some typing. As an added bonus, you might even find the `dplyr` grammar easier to read.
+
+Here we're going to cover 6 of the most commonly used functions as well as using pipes (`%>%`) to combine them. 
+
+1. `select()`
+2. `filter()`
+3. `group_by()`
+4. `summarize()`
+5. `mutate()`
+
+If you have have not installed this package earlier, please do so:
+
+```{r,eval=FALSE}
+install.packages('dplyr')
+```
+
+Now let's load the package:
+
+```{r,message=FALSE}
+library(dplyr)
+```
+
+#### Using select()
+
+If, for example, we wanted to move forward with only a few of the variables in our dataframe we could use the `select()` function. This will keep only the variables you select.
+
+```{r}
+year_country_gdp <- select(gapminder,year,country,gdpPercap)
+```
+
+```{r,echo=FALSE}
+library(DiagrammeR)
+grViz('digraph html {
```

Comment:
I think so... otherwise it is a lot of material to scan through... `source(figure1.R)` etc would be a lot more concise. But I am not sure what best practice should be. Maybe @gvwilson or @jdblischak could weigh in?